arch/arc/core/regular_irq.S

```diff
@@ -35,6 +35,168 @@ GTEXT(_rirq_common_interrupt_swap)
  * TODO: Revist this if FIRQ becomes configurable.
  */
 
+/*
+
+===========================================================
+ RETURN FROM INTERRUPT TO COOPERATIVE THREAD
+===========================================================
+
+That's a special case because:
+ 1. We return from IRQ handler to a cooperative thread
+ 2. During IRQ handling context switch did happen
+ 3. Thread we're returining to previously voluntarily gave control
+    to another thread because of:
+     - Falling k_sleep()
```

Comment:
Calling, not Falling, I assume

Reply:
I meant exactly "falling" in a context "falling asleep", this one even sounds similarly "falling ka-sleep" :)

Reply:
yeah, I though about that. But, falling could also be mistaken/misread for failing...